### ISSUE TITLE
Vehicles stopped driving on green light in the middle of the simulation and stay on lane

### ISSUE THREAD

**Issue by lola-notraffic** [eclipse/sumo#9857]
Hi,
I'm running scenario of 1 hour duration in SUMO with generated vehicles in the same pattern each run. After approx. 40 min of simulation there is a lane where vehicles suddenly stopped driving on the green light and just staying on the lane till the end of the simulation.
Any ideas what can  cause it?

Thank you,
Olga  
---
SUMO-version: 1.9.2
operating system: Ubuntu 20.04.3

**Comment by namdre** [comment]
There are only a few reasons for this (beyond this being a bug).
Possibly you have loaded vehicles with invalid routes and set option --ignore-route-errors.

Are you using TraCI to influence the simulation?
What options are in use?
Did you check the route of the vehicle in question? (https://sumo.dlr.de/docs/sumo-gui.html#showing_routes_and_route-related_information)

**Comment by lola-notraffic** [comment]
Thank you for your quick response!
I'm using Traci for simulation yes.
There no special options I'm using. 
I will try --ignore-route-errors as you suggested. Should I also see invalid routes in sumo gui logs?
Thank you again!

**Comment by namdre** [comment]
I mentioned the option --ignore-route-errors as a possible reason for the misbehavior you are seeing. If the option isn't set, the misbehavior has a different cause and setting the option will not help you.
Are you calling traci commands that change vehicle behavior (i.e. vehicle.setSpeed?)

**Comment by lola-notraffic** [comment]
Hi again.
I'm only using vehicle.setType and different vehicle.get... methods. There are other scenarios that are working with the same type of vehicle that I set. 
But I've checked if there are any problematic routes that I have maybe created. I found some routes that were created on the wrong lanes (from what I saw the problem is in my model and that is why the vehicles are created in wrong lanes: i.e. left turn route is created on the lane that goes straight). So I'm currently investigating it.
Thank you again! I'll update you when I'm finished with checking the issue I've talked about.
